### PR TITLE
Added MySQL DB link

### PR DESCRIPTION
**Pull request by dxminic**

### CODE REVIEW

File: src/ch/jamiete/hilda/HildaDB.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+public class HildaDB {
+    private static final String DATABASE_DRIVER = "com.mysql.jdbc.Driver";
+    private static final String DATABASE_URL = "";
+    private static final String USERNAME = "";
+    private static final String PASSWORD = "";
+    private static final String MAX_POOL = "250";
+
+    private Connection con;
+    private Properties properties;
```
Comment (jamietech) [inline src/ch/jamiete/hilda/HildaDB.java:15]: Can this not be stored only in the strings and then used with the JDBC methods instead of making a properties file?

File: src/ch/jamiete/hilda/HildaDB.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+            } catch (SQLException e) {
+                Hilda.getLogger().log(Level.SEVERE, "Error has occurred when disconnecting from the database:", e);
+            }
+        }
+    }
+
+    public void updateDb(String s) {
+        try {
+            PreparedStatement pst;
+            pst = this.connect().prepareStatement("insert into messages (messagescol)" + "VALUES (?)");
```
Comment (jamietech) [inline src/ch/jamiete/hilda/HildaDB.java:54]: What is this?

File: src/ch/jamiete/hilda/HildaDB.java [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+        if (properties == null) {
+            properties = new Properties();
+            properties.setProperty("user", USERNAME);
+            properties.setProperty("password", PASSWORD);
+            properties.setProperty("MaxPooledStatements", MAX_POOL);
+        }
+        return properties;
+    }
+
+    Connection connect() {
```
Comment (jamietech) [inline src/ch/jamiete/hilda/HildaDB.java:27]: Should we not use HikariCP?

File: src/ch/jamiete/hilda/HildaDB.java [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+            pst = this.connect().prepareStatement("insert into messages (messagescol)" + "VALUES (?)");
+            pst.setString(1,s);
+            pst.executeUpdate();
+        } catch (SQLException e) {
+            Hilda.getLogger().log(Level.SEVERE, "Error has occurred when updating the database:", e);
+        } finally {
+            this.disconnect();
+        }
+    }
+}
```
Comment (jamietech) [inline src/ch/jamiete/hilda/HildaDB.java:63]: There should be a method to generate prepared statements.

File: src/ch/jamiete/hilda/Hilda.java [hunk 5]
```diff
@@ -168,11 +145,13 @@ public void start() {
         this.plugins.enablePlugins();
         Hilda.getLogger().info("Plugins enabled!");
 
+        Hilda.getLogger().info("Connecting to database...");
+        this.db.connect();
```
Comment (jamietech) [inline src/ch/jamiete/hilda/Hilda.java:149]: There needs to be a connected log

File: src/ch/jamiete/hilda/Hilda.java [hunk 6]
```diff
@@ -143,7 +119,7 @@ public void start() {
         this.executor.setRemoveOnCancelPolicy(true);
         this.executor.setMaximumPoolSize(10);
 
-        final long rotate = this.getNextMidnightInMillis("GMT+10") - System.currentTimeMillis();
+        final long rotate = Util.getNextMidnightInMillis("GMT+10") - System.currentTimeMillis();
```
Comment (jamietech) [inline src/ch/jamiete/hilda/Hilda.java:122]: This should probably be in a separate commit

File: src/ch/jamiete/hilda/Hilda.java [hunk 7]
```diff
@@ -133,7 +109,7 @@ public String getUsername() {
         return this.bot.getSelfUser().getName();
     }
 
-    public void start() {
+    void start() {
```
Comment (jamietech) [inline src/ch/jamiete/hilda/Hilda.java:112]: You should not have removed the scope

**Review by jamietech — CHANGES_REQUESTED** [review]

File: src/ch/jamiete/hilda/Hilda.java (hunk 7 above)
Comment (dxminic) [reply src/ch/jamiete/hilda/Hilda.java:112]: Please see https://stackoverflow.com/questions/215497/in-java-difference-between-package-private-public-protected-and-private.

File: src/ch/jamiete/hilda/Hilda.java (hunk 6 above)
Comment (dxminic) [reply src/ch/jamiete/hilda/Hilda.java:122]: Saves time. Multiple commit clutters git log,

File: src/ch/jamiete/hilda/Hilda.java (hunk 5 above)
Comment (dxminic) [reply src/ch/jamiete/hilda/Hilda.java:149]: There is in the DB class.

File: src/ch/jamiete/hilda/HildaDB.java (hunk 1 above)
Comment (dxminic) [reply src/ch/jamiete/hilda/HildaDB.java:15]: No.

File: src/ch/jamiete/hilda/HildaDB.java (hunk 3 above)
Comment (dxminic) [reply src/ch/jamiete/hilda/HildaDB.java:27]: No.

File: src/ch/jamiete/hilda/HildaDB.java (hunk 2 above)
Comment (dxminic) [reply src/ch/jamiete/hilda/HildaDB.java:54]: SQL.

File: src/ch/jamiete/hilda/HildaDB.java (hunk 4 above)
Comment (dxminic) [reply src/ch/jamiete/hilda/HildaDB.java:63]: There is. (updateDb)

File: src/ch/jamiete/hilda/Hilda.java (hunk 7 above)
Comment (jamietech) [reply src/ch/jamiete/hilda/Hilda.java]: If you are trying to change the scope of methods in the main file you should do so in a different commit.

File: src/ch/jamiete/hilda/Hilda.java (hunk 6 above)
Comment (jamietech) [reply src/ch/jamiete/hilda/Hilda.java]: The git log will survive another day.

File: src/ch/jamiete/hilda/Hilda.java (hunk 5 above)
Comment (jamietech) [reply src/ch/jamiete/hilda/Hilda.java:149]: It should be in this class as is convention.

Comment (dxminic) [reply src/ch/jamiete/hilda/Hilda.java:149]: But should it fail to connect due to a SQLException we do not want it to log connected, as a result it should be logged in the DB class. As in convention, see your plugins.PluginManager class.

File: src/ch/jamiete/hilda/HildaDB.java (hunk 1 above)
Comment (jamietech) [reply src/ch/jamiete/hilda/HildaDB.java:15]: https://docs.oracle.com/javase/8/docs/api/java/sql/DriverManager.html#getConnection-java.lang.String-java.lang.String-java.lang.String-

**Comment by jamietech** [comment]
Thank you for your pull request. Unfortunately, it will not be merged unless you fix the:

1. Conflicts
2. Review requests
3. Extraneous commits meant for other PRs
